### PR TITLE
samples: drivers: counter: alarm: add nordic boards to the whitelist

### PR DESCRIPTION
**Pull request by nordic-krch**
Add nordic boards to the whitelist to ensure that compilation is
checked by the CI.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

File: samples/drivers/counter/alarm/sample.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ tests:
   sample.driver.counter.alarm:
     tags: drivers
     harness: console
-    platform_whitelist: nucleo_f746zg
+    platform_whitelist: nucleo_f746zg nrf52840_pca10056 nrf9160_pca10090
```
Comment (ioannisg) [inline samples/drivers/counter/alarm/sample.yaml]: Why not the pca10040, @nordic-krch ?

Comment (ioannisg) [reply samples/drivers/counter/alarm/sample.yaml]: or the pca10028

Comment (nordic-krch) [reply samples/drivers/counter/alarm/sample.yaml]: yeah, i can add them as well. wanted to add at least from each series.

Comment (ioannisg) [reply samples/drivers/counter/alarm/sample.yaml]: Yeah, please add the nrf51, if applicable, and perhaps nrf52_pca10040, as this is the "mainstream" 52.

Comment (nordic-krch) [reply samples/drivers/counter/alarm/sample.yaml]: done

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**